docs/developer/community/zero-downtime-migrations.mdx

```diff
@@ -0,0 +1,344 @@
+- task should que yourself if there is still data to be processed
+- task should proceed with data from newest to oldest
+
+```mermaid
+graph TD
+    subgraph Migraion
+        A[Migration trigger task in post-migrate if there is any object in the table.]
+    end
+    subgraph Task
+        A -->B{{If there is any object to update?}}
+        B -->|Yes| C[Take X of the latest object to update..]
+        C -->D[Update selected objects.]
+        D --> E[Trigger again same update tasks.]
+        E -->B
+        B -->|No|F{{Is there a next task to trigger e.g. Data migration to the next model?}}
+        F -->|Yes|G[Trigger next task.]
+        F -->|No|H[End]
+    end
+```
+
+e.g. from Saleor core code, task to change `type` in `OrderEvent` from `transaction_void_requested` to `transaction_cancel_requested`.
+
+```python
+from ....celeryconf import app
+from ...models import OrderEvent
+
+from django.db import transaction
+from django.db.models import QuerySet
+
+# batch size to make sure that task is completed in 1 second and as well we don't use too much memory
+BATCH_SIZE = 5000
+
+
+def update_type_to_transaction_cancel_requested(qs: QuerySet[OrderEvent]):
+    with transaction.atomic():
+        # lock the batch of objects, to avoid deadlocks
+        _events = list(qs.select_for_update(of=(["self"])))
+        qs.update(type="transaction_cancel_requested")
+
+
+@app.task
+def order_events_rename_transaction_void_events_task():
+    # Order events proceed from the newest to the oldest
+    events = OrderEvent.objects.filter(type="transaction_void_requested").order_by(
+        "-pk"
+    )
+    ids = events.values_list("pk", flat=True)[:BATCH_SIZE]
+    qs = OrderEvent.objects.filter(pk__in=ids)
+
+    # If we found data, queue next execution of the task
+    if ids:
+        update_type_to_transaction_cancel_requested(qs)
+        order_events_rename_transaction_void_events_task.delay()
+```
+
+## Upgrading with zero-downtime migrations
+
+For the general upgrade guide see the [upgrading guide](../../setup/upgrading).
+Zero-downtime migrations can only be achieved by upgrading Saleor minor version by minor version, that means, to upgrade Saleor from version 3.11 to 3.13, you need first upgrade your workers
+to version 3.12, then proceed with upgrading to version 3.13. By upgrading from version 3.11 directly to 3.13 you will not follow zero-downtime policy.
+
+The exact steps needed to be done:
+
+- Upgrade your current minor version to the newest patch of Saleor of that version, remember to turn off Celery workers as stated in [upgrading guide](../../setup/upgrading).
+- Upgrade your current minor version of Saleor to the next one. Remember to follow the steps stated in [upgrading guide](../../setup/upgrading).
+
+### Troubleshooting: Migration with celery task fails
+
+It's safe to rerun failed migration tasks in case of an error. To do that you can for example start a task from `Django shell`
```

Comment:
```suggestion
It's safe to rerun failed migration tasks in case of an error. To do that, you can for example start a task from the Django shell:
```